app/scripts/modules/google/serverGroup/configure/serverGroupCommandBuilder.service.js

```diff
@@ -176,18 +176,40 @@ module.exports = angular.module('spinnaker.gce.serverGroupCommandBuilder.service
     function populateCustomMetadata(metadataItems, command) {
       // Hide metadata items in the wizard.
       if (metadataItems) {
+        let customUserData = '';
+        let customUserDataKeys = [];
         if (angular.isArray(metadataItems)) {
           metadataItems.forEach(function (metadataItem) {
-            if (!_.includes(gceServerGroupHiddenMetadataKeys, metadataItem.key)) {
+            if (metadataItem.key === 'customUserData') {
```

Comment:
I would try to find `customUserData` (and determine `customUserDataKeys`) inside `metadataItems` before looping over it - that way you can guarantee that you'll omit those keys from `command.instanceMetadata`.